### DISCUSSION TITLE
How to hide all UI window elements in Sparkle?

### DISCUSSION THREAD

**Discussion by nishaanth-ms-21155** [sparkle-project/Sparkle#2623] [Q&A]
I have managed to block AutoUpdate.app passsword prompt while relaunching by changing `checkForUpdatesinBackgound` to `checkForUpdatesinBackgoundWithoutUI` But it desn't seem to block New Update is Available window with Release notes shown.

**Comment by zorgiepoo** [comment]
> How to hide all UI window elements in Sparkle? 

You can minimize the UI shown in many circumstances but you cannot always hide the UI. The reason for this is because some user's setup may require them to authenticate with their user name / password in order to install an update (depending on where the app is installed and its ownership). There must be UI in that case otherwise it will be a buggy user experience.

And the way Sparkle supports that is to enable automatic downloading/installing of updates -- see [SUAutomaticallyUpdate](https://sparkle-project.org/documentation/customization/). Updates are silently installed on quit as long as the user doesn't need to authenticate. By default, Sparkle may prompt the user if a new update is waiting to be installed and the user hasn't quit the app for 1 week though.

> checkForUpdatesinBackgoundWithoutUI

There's no such method, and this is not the right route.

(Autoupdate itself doesn't manage showing the prompt in Sparkle 2, but that is an implementation detail.)
